src/workspacefactory/wfactory_controller.js

```diff
@@ -102,6 +107,65 @@ WorkspaceFactoryController.MODE_JS = 'javascript';
 WorkspaceFactoryController.MODE_XML = 'xml';
 
 /**
+ * Creates a new toolbox for editing. Saves previously edited toolbox, and prompts
+ * user if previous toolbox has not been saved under a user-specified name. Initial
+ * default name of toolbox is the empty string, and is considered an un-named
+ * toolbox that the user will eventually rename if they add another toolbox.
+ */
+WorkspaceFactoryController.prototype.newToolbox = function() {
+  // TODO: implement
+  // Prompt user for name of new toolbox.
+  // Check if name is valid / taken.
+  // If taken, prompt again.
+  // If not taken, model.addToolbox(newName).
+  // Show toolbox after successful naming.
+};
+
+/**
+ * Saves XML currently in workspace into currently active toolbox under this.toolboxList.
+ *
+ * @returns {boolean} If saved successfully.
+ */
+WorkspaceFactoryController.prototype.saveToolbox = function() {
+  // TODO: implement
+  // Check if current toolbox has a name (model.ifNamedToolbox()).
+  // If no name, prompt user to name current toolbox.
+  // If named, model.updateToolbox(model.currentToolbox).
+};
+
+/**
+ * Changes view to display a different or new toolbox to edit.
+ *
+ * @param {string} name Name of toolbox to display.
+ * @returns {boolean} If toolbox name was valid and displayed succesfully onto
+ *     workspace.
```

Comment:
Because this may prompt for user feedback, the success may not be know immediately. Consider using a callback or a Promise to return the success status.